ray_beam_runner/portability/state.py

```diff
@@ -19,12 +19,40 @@
 
 import collections
 import contextlib
-from typing import Optional, Tuple, Iterator
+from typing import Optional, Tuple, Iterator, TypeVar
 
 import ray
+from ray import ObjectRef
 from apache_beam.portability.api import beam_fn_api_pb2
 from apache_beam.runners.worker import sdk_worker
 
+T = TypeVar("T")
+
+
+class RayFuture(sdk_worker._Future[T]):
```

Comment:
Yes, I think that's right. One thing that's a bit unfortunate about this adapter is that the SdkWorker waits for a series of these futures in sequence, which ray can do natively (with `ray.get(list_of_futures)`). I'm not sure whether it will actually matter in this case, though.